modules/markets-api/src/main/scala/org/ergoplatform/dex/markets/services/FiatRates.scala

```diff
@@ -1,29 +1,32 @@
 package org.ergoplatform.dex.markets.services
 
-import cats.effect.Clock
-import cats.{FlatMap, Functor, Monad}
-import derevo.derive
-import org.ergoplatform.common.caching.Memoize
+import cats.effect.concurrent.Ref
+import cats.effect.{Clock, Sync, Timer}
+import cats.syntax.option._
+import cats.{FlatMap, Monad}
 import org.ergoplatform.dex.domain.{AssetClass, FiatUnits}
 import org.ergoplatform.dex.markets.currencies.UsdUnits
 import org.ergoplatform.dex.protocol.constants.{ErgoAssetClass, ErgoAssetDecimals}
-import org.ergoplatform.ergo.domain.{RegisterId, SConstant}
 import org.ergoplatform.ergo.TokenId
+import org.ergoplatform.ergo.domain.{RegisterId, SConstant}
 import org.ergoplatform.ergo.services.explorer.ErgoExplorer
+import tofu.Catches
 import tofu.concurrent.MakeRef
-import tofu.higherKind.Mid
-import tofu.higherKind.derived.representableK
 import tofu.logging.{Logging, Logs}
+import tofu.streams.Evals
 import tofu.syntax.foption._
+import tofu.syntax.handle._
 import tofu.syntax.logging._
 import tofu.syntax.monadic._
+import tofu.syntax.streams.evals._
 
 import scala.concurrent.duration._
 
-@derive(representableK)
-trait FiatRates[F[_]] {
+trait FiatRates[F[_], S[_]] {
 
   def rateOf(asset: AssetClass, units: FiatUnits): F[Option[BigDecimal]]
+
+  def run: S[Unit]
 }
```

Comment:
Wtf? Why FiatRates a process and a service at the same time?

Reply:
reworked